autoit_ripper/autoit_unpack.py

```diff
@@ -1,302 +1,147 @@
 import logging
-import struct
-from enum import Enum
-from typing import Optional, Tuple, Iterator, List
+from typing import Any, Iterator, List, Optional, Tuple, Union
 
-import lief  # type: ignore
+import pefile  # type: ignore
 
-from .opcodes import OPCODES
-from .utils import BitStream, crc_data, decrypt_lame, decrypt_mt, filetime_to_dt
-
-EA05_MAGIC = bytes.fromhex("a3484bbe986c4aa9994c530a86d6487d41553321")
-
-
-class AutoItVersion(Enum):
-    EA05 = 0
-    EA06 = 1
-
-
-# 10 megabytes
-MAX_SCRIPT_SIZE = 10 * 10 ** 6
+from .decompress import decompress
+from .opcodes import deassemble_script
+from .utils import (
+    AutoItVersion,
+    ByteStream,
+    EA05Decryptor,
+    EA06Decryptor,
+    crc_data,
+    filetime_to_dt,
+)
 
 log = logging.getLogger(__name__)
 
 
-def u32(data: bytes) -> int:
-    return struct.unpack("<I", data[:4])[0]
-
-
-def get_script_resource(pe: lief.PE) -> Optional[lief.PE.ResourceDirectory]:
-    for child in pe.resources.childs:
-        for grandchild in child.childs:
-            if grandchild.has_name and grandchild.name == "SCRIPT":
-                return grandchild
-    return None
+EA05_MAGIC = bytes.fromhex("a3484bbe986c4aa9994c530a86d6487d41553321")
 
 
-def decompress(data: bytes, version: AutoItVersion) -> Optional[bytes]:
-    if data[:4] == b"EA05" and version == AutoItVersion.EA05:
-        pass
-    elif data[:4] == b"EA06" and version == AutoItVersion.EA06:
-        pass
+def find_root_dir(pe: pefile.PE, RT_Name: str) -> Optional[pefile.ResourceDirData]:
+    dir_entries = [
+        entry
+        for entry in pe.DIRECTORY_ENTRY_RESOURCE.entries
+        if entry.id == pefile.RESOURCE_TYPE[RT_Name]
+    ]
+    if len(dir_entries) == 1:
+        return dir_entries[0].directory
+    elif len(dir_entries) == 0:
+        log.error("Couldn't find any appropiate PE resource directory")
+        return None
     else:
-        log.error("Magic mismatch: %s", data[:4])
+        log.error("Couldn't find any appropiate PE resource directory")
```

Comment:
Error messages here are not very helpful and duplicated. Maybe change to `else` and `f"Expected unique resource named {RT_Name}, found {len(entries)}"`?